include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -2495,7 +2495,7 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::integrate_fast(
              is_linear ? (is_face_path ? 2 * Utilities::pow(2, dim - 1) :
                                          Utilities::pow(2, dim)) :
                          0>
-    solution_values_vectorized_linear;
+    solution_values_vectorized_linear = {};
```

Comment:
Is this actually adding instructions to zero the fields in the `is_linear` path, or is the compiler able to remove the initialization? It would be good if there is no additional work done in release mode, as it in fact only pertains to a compiler behavior?

Reply:
Are the instructions only some `vxorpd` instructions (which are cheap, and even eliminated on some architectures because the processor might recognize it as zeroing idiom, https://stackoverflow.com/questions/33666617/what-is-the-best-way-to-set-a-register-to-zero-in-x86-assembly-xor-mov-or-and), or is the zero result moved back to memory? In the former case, I guess we can ignore the issue, in the latter we need to think about another solution

Reply:
```
   61ad9:       c5 c9 57 f6             vxorpd %xmm6,%xmm6,%xmm6
   61add:       45 89 cb                mov    %r9d,%r11d
   61ae0:       48 8b 8f 40 01 00 00    mov    0x140(%rdi),%rcx
   61ae7:       41 83 e3 01             and    $0x1,%r11d
   61aeb:       41 83 e1 02             and    $0x2,%r9d
   61aef:       45 8d 56 ff             lea    -0x1(%r14),%r10d
   61af3:       31 c0                   xor    %eax,%eax
   61af5:       31 d2                   xor    %edx,%edx
   61af7:       c5 fd 28 e6             vmovapd %ymm6,%ymm4
   61afb:       c5 7d 28 c6             vmovapd %ymm6,%ymm8
   61aff:       c5 fd 28 ee             vmovapd %ymm6,%ymm5
   61b03:       c5 fd 28 fe             vmovapd %ymm6,%ymm7
   61b07:       c5 fd 29 74 24 c8       vmovapd %ymm6,-0x38(%rsp)
   61b0d:       c5 fd 29 74 24 e8       vmovapd %ymm6,-0x18(%rsp)
   61b13:       c5 fd 29 74 24 a8       vmovapd %ymm6,-0x58(%rsp)
   61b19:       e9 b3 00 00 00          jmpq   61bd1 <_ZN6dealii17FEPointEvaluationILi1ELi3ELi3ENS_15VectorizedArrayIdLm4EEEE14integrate_fastILb1ELb0ELb1ELm4EEEvRKNS_16StridedArrayViewIdXT2_EEERKNS_15EvaluationFlags15EvaluationFlagsE+0x271>
   61b1e:       66 90                   xchg   %ax,%ax
   61b20:       c5 fd 28 44 01 40       vmovapd 0x40(%rcx,%rax,1),%ymm0
   61b26:       c5 fd 28 0c 01          vmovapd (%rcx,%rax,1),%ymm1
   61b2b:       c5 fd 59 d3             vmulpd %ymm3,%ymm0,%ymm2
   61b2f:       c5 7d 28 4c 01 20       vmovapd 0x20(%rcx,%rax,1),%ymm9
   61b35:       45 85 c9                test   %r9d,%r9d
   61b38:       0f 85 ca 00 00 00       jne    61c08 <_ZN6dealii17FEPointEvaluationILi1ELi3ELi3ENS_15VectorizedArrayIdLm4EEEE14integrate_fastILb1ELb0ELb1ELm4EEEvRKNS_16StridedArrayViewIdXT2_EEERKNS_15EvaluationFlags15EvaluationFlagsE+0x2a8>
   61b3e:       c5 7d 28 2d fa d2 04    vmovapd 0x4d2fa(%rip),%ymm13        # aee40 <_ZZNSt19_Sp_make_shared_tag5_S_tiEvE5__tag+0x78>
```
If I interpret it correctly it uses `vxorpd` in `61ad9` and then uses all available vector registers like in `61af7` and then moves the rest on the stack like in `61b07`. Callgrind tells me with initialization we need 7 additional instructions per call.

Reply:
Yes, it looks like 3 registers need to be written back to memory, the rest remains in the registers. It is indeed unfortunate, I would not have cared much about `61af7` to `61b03`. Can you measure it in the performance profiles as well?

Regarding alternatives, I looked at the uses. Obviously, the compiler should see that all cases do initialize it, so it seems it lost track about where the variable is set versus used. If you have a system that exposes this, you could try to see if you can narrow it down a bit, e.g. by moving the finalization closer to the definition, but I am not sure it helps without breaking modularity.

Reply:
At least on my workstation, I don't see a visible decrease in performance for this or #16022 compared to master. I can give both a spin on our server tomorrow.